packages/cache/src/internal/downloadUtils.ts

```diff
@@ -242,7 +242,9 @@ export async function downloadCacheStorageSDK(
     // If the file exceeds the buffer maximum length (~1 GB on 32-bit systems and ~2 GB
     // on 64-bit systems), split the download into multiple segments
     // ~2 GB = 2147483647, beyond this, we start getting out of range error. So, capping it accordingly.
-    const maxSegmentSize = Math.min(2147483647, buffer.constants.MAX_LENGTH)
+
+    // Updated segment size to 128MB = 134217728 bytes, to complete a segment faster and fail fast
+    const maxSegmentSize = Math.min(134217728, buffer.constants.MAX_LENGTH)
```

Comment:
Should it be one less to keep it similar as earlier?

Reply:
I thought of it too, but we had done it for 2GB not to cross the limit. Here 128 is safe. So every segment will be complete 128MB and not one byte less, which I thought shouldn't be a problem?

Reply:
Sounds good!